pkg/operator/encryption/controllers/condition_controller.go

```diff
@@ -54,26 +45,25 @@ func NewConditionController(
 	encryptionSecretSelector metav1.ListOptions,
 	eventRecorder events.Recorder,
 	encryptedGRs []schema.GroupResource,
-) *conditionController {
+) factory.Controller {
 	c := &conditionController{
 		operatorClient: operatorClient,
 
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EncryptionConditionController"),
-		eventRecorder: eventRecorder.WithComponentSuffix("encryption-condition-controller"),
-
 		encryptedGRs: encryptedGRs,
 
 		encryptionSecretSelector: encryptionSecretSelector,
 		deployer:                 deployer,
 		secretClient:             secretClient,
 	}
 
-	c.preRunCachesSynced = setUpInformers(deployer, operatorClient, kubeInformersForNamespaces, c.eventHandler())
-
-	return c
+	return factory.New().WithInformers(
+		kubeInformersForNamespaces.InformersFor("openshift-config-managed").Core().V1().Secrets().Informer(),
+		operatorClient.Informer(),
+		deployer,
+	).ResyncEvery(time.Second).WithSync(c.sync).ToController("EncryptionConditionController", eventRecorder.WithComponentSuffix("encryption-condition-controller"))
```

Comment:
resync every second?

Reply:
@sttts that was pre-existing https://github.com/openshift/library-go/pull/687/files/318a3bf613627b8148ed47788882d3f1a1490705#diff-a03dd41db6b9c24ab2e28afc2b14fe5dL199

Reply:
wow. Sounds like we would prefer something like 30 sec at least.

Reply:
@sttts this is all pre-existing... i would do this as a follow up maybe?

Reply:
follow-up is fine